### PR TITLE
Added aseries for MRV_Asymptotic algorithm

### PR DESCRIPTION
**Pull request by arighnac**
Implementation of the **MRV Asympt** algorithm, for asymptotic expansions( i,e, asymptotic computations of series). The Idea can be found in my [GSoC proposal](https://github.com/sympy/sympy/wiki/GSoC-2018-Application-Arighna-Chakrabarty-:-Improving-Series-Expansions). 

More details about this can be found [here](https://dl.acm.org/citation.cfm?id=164136). All the test cases have been added, and implemented.

### CODE REVIEW

**Comment by asmeurer** [comment]
Please add some tests

**Comment by asmeurer** [comment]
Is this related to issue #14563?

File: sympy/core/expr.py [hunk 1]
```diff
@@ -2877,6 +2877,57 @@ def _eval_nseries(self, x, n, logx):
                      nseries calls it.""" % self.func)
                      )
 
+    def aseries(self, x, n=6, bound=0, hierar=False):
+        from sympy.series.gruntz import mrv, rewrite, mrv_leadterm
+        from sympy.functions import exp, log
+
+        if x.is_positive is x.is_negative is None:
+            xpos = C.Dummy('x', positive=True, bounded=True)
```
Comment (asmeurer) [inline sympy/core/expr.py]: C is not defined here. If this is supposed to be the class registry, that is deprecated (just import Dummy).

File: sympy/core/expr.py [hunk 2]
```diff
@@ -2877,6 +2877,57 @@ def _eval_nseries(self, x, n, logx):
...
+            return self.subs(x, xpos).aseries(xpos, n, bound, hir).subs(xpos, x)
+
+        omega, exps = mrv(self, x)
+        if x in omega:
+            s = self.subs(x, exp(x)).aseries(x, n, bound, hir).subs(x, log(x))
```
Comment (asmeurer) [inline sympy/core/expr.py]: `hir` is undefined here.

**Comment by ArighnaIITG** [comment]
@asmeurer , This PR is not at all related to issue #14563 .

This is a PR which intends to apply the MRV_Asympt Algorithm, for the evaluation of asymptotic series. It is a part of my GSoC proposal, and I tried to start the work beforehand.

All the test cases from "Gruntz thesis" have been implemented here.
Please review.

@jksuom , @smichr .

File: sympy/core/expr.py (hunk 2 above)
Comment (ArighnaIITG) [reply sympy/core/expr.py]: Sorry for the late response. 
Have updated everything, and also added tests. ( See the comment below).

**Comment by skirpichev** [comment]
> Is this related to issue #14563?

This is "related" to #7529.

**Comment by asmeurer** [comment]
@ArighnaIITG if you are reusing code from an older pull request, you should reuse the original commits so that the authorship remains intact. You should also make a note of this in the pull request description (thanks for pointing this out @skirpichev).

**Comment by Abdullahjavednesar** [comment]
There's already a PR with exactly the same changes. I'm labeling this duplicate,

**Comment by ArighnaIITG** [comment]
Yes, @asmeurer , I will do that. I was actually not aware of the process.

**Pull request closed without merging**